pandas/tests/scalar/timedelta/test_arithmetic.py

```diff
@@ -616,3 +690,35 @@ def test_rdivmod_invalid(self):
 
         with pytest.raises(TypeError):
             divmod(np.array([22, 24]), td)
+
+    def test_td_div_timedelta_timedeltalike_array(self):
+        # GH 21980
+        arr = np.array([Timestamp('20130101 9:01'),
+                        Timestamp('20121230 9:02')])
+
+        with pytest.raises(TypeError):
+            arr / pd.Timedelta('1D')
+
+    def test_td_rdiv_timedelta_mixed_timedeltalike_array(self):
+        # GH 21980
+        arr = np.array([pd.Timestamp.now(),
+                        pd.Timedelta('1D')])
+
+        with pytest.raises(TypeError):
+            pd.Timedelta('1D') / arr
+
+    def test_td_mult_timedelta_mixed_timedeltalike_array(self):
+        # GH 21980
+        arr = np.array([pd.Timestamp.now(),
+                        pd.Timedelta('1D')])
+
+        with pytest.raises(TypeError):
+            pd.Timedelta('1D') * arr
+
+    def test_td_rmult_timedelta_mixed_timedeltalike_array(self):
```

Comment:
Some of these you can probably de-duplicate using `pytest.mark.parametrize`.  Not a deal-breaker.

If there's a graceful way to work "object_dtype" into the test names that'd be ideal.

Reply:
It might be a lot more effor to get this de-duped than I initially thought, would it be ok to submit as-is? I added in object_dtype to the method titles.

Reply:
pls parameterize in this PR.

Reply:
@illegalnumbers the idea here is to notice that all four of these tests are special cases of a single test function.  In particular, you can write `arr * pd.Timedelta('1D')` as `operator.mul(arr, pd.Timedelta('1D')` and similarly for the others with `operator.truediv`, `ops.rdiv`, `ops.rmul`.  Then these can be parametrized as:

```
@pytest.mark.parametrize('op', [operator.mul, ops.rmul, operator.truediv, ops.rdiv])
def test_td_mul_object_array(self, op):
    arr = np.array([pd.Timestamp.now(), pd.Timedelta('1D')])

    with pytest.raises(TypeError):
        op(arr, pd.Timedelta('1D'))
```